code/modules/library/random_books.dm

```diff
@@ -18,31 +29,36 @@
 	if(!book_count || !isnum(book_count))
 		update_icon()
 		return
+	book_count += pick(-1,-1,0,1,1)
+	create_random_books(book_count, src, FALSE, category)
+	update_icon()
+
+/proc/create_random_books(amount = 2, location, fail_loud = FALSE, category = null)
+	. = list()
+	if(!isnum(amount) || amount<1)
+		return
 	if(!establish_db_connection())
-		if(prob(5))
-			var/obj/item/weapon/paper/P = new(get_turf(loc))
+		if(fail_loud || prob(5))
+			var/obj/item/weapon/paper/P = new(location)
 			P.info = "There once was a book from Nantucket<br>But the database failed us, so f*$! it.<br>I tried to be good to you<br>Now this is an I.O.U<br>If you're feeling entitled, well, stuff it!<br><br><font color='gray'>~</font>"
-		update_icon()
-		return
-	if(category && prob(25)) category = null
-	book_count += pick(-1,-1,0,1,1)
-	if(book_count <= 0)
-		update_icon()
+			P.update_icon()
 		return
-
+	if(prob(25))

```

Comment:
It was there before.  I assume it's so that occasionally bookshelves aren't stocked with the right books, to further enhance the "Nanotrasen doesn't care" look.